test/CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ ExternalProject_Add(
     GIT_REPOSITORY "https://github.com/google/googletest.git"
     SOURCE_DIR "${CMAKE_SOURCE_DIR}/../vendor/googletest"
```

Comment:
can we change this to be inside the BUILD_DIR instead of the SOURCE_DIR? Especially if we stick with this mechanism for a while we should not make changes to the SOURCE_DIR.